packages/compose/src/hooks/use-focus-on-mount/index.js

```diff
@@ -79,6 +79,23 @@ export default function useFocusOnMount( focusOnMount = 'firstElement' ) {
 			return;
 		}
 
+		if ( typeof focusOnMountRef?.current === 'function' ) {
+			timerId.current = setTimeout( () => {
+				const tabbables = focus.tabbable.find( node );
+
+				// Ignoring next line because the type is actually callable,
+				// but the linter cannot recognise the outer typeof check.
+				// @ts-ignore
+				const elementToFocus = focusOnMountRef.current( tabbables );
```

Comment:
Another option to appease the linter is to wrap with another `typeof` check:

```suggestion
				// Ignoring next line because the type is actually callable,
				// but the linter cannot recognise the outer typeof check.
				// @ts-ignore
				if ( typeof focusOnMountRef?.current === 'function' ) {
				    const elementToFocus = focusOnMountRef.current( tabbables );
				}
```

Reply:
Do we still the disabling comment then?

Reply:
We only need the disable _if_ we update the PR to include the additional `typeof` check. I just thought it was superfluous code and would be better achieved by telling the linter to ignore rather than add unnecessary bytes.

Maybe some TS gurus can put be right though?

Reply:
I think that TS may be correct because the above check cannot ensure that the `focusOnMountRef` will still hold a reference to a function when the timeout will fire.

One way to get around it is to grab the value of `focusOnMountRef.current` before the `setTimeout`:

```diff
diff --git a/packages/compose/src/hooks/use-focus-on-mount/index.js b/packages/compose/src/hooks/use-focus-on-mount/index.js
index b300a2a9a0..e9a3e5faa2 100644
--- a/packages/compose/src/hooks/use-focus-on-mount/index.js
+++ b/packages/compose/src/hooks/use-focus-on-mount/index.js
@@ -79,14 +79,12 @@ export default function useFocusOnMount( focusOnMount = 'firstElement' ) {
 			return;
 		}
 
-		if ( typeof focusOnMountRef?.current === 'function' ) {
+		if ( typeof focusOnMountRef.current === 'function' ) {
+			const fn = focusOnMountRef.current;
 			timerId.current = setTimeout( () => {
 				const tabbables = focus.tabbable.find( node );
 
-				// Ignoring next line because the type is actually callable,
-				// but the linter cannot recognise the outer typeof check.
-				// @ts-ignore
-				const elementToFocus = focusOnMountRef.current( tabbables );
+				const elementToFocus = fn( tabbables );
 
 				if ( elementToFocus ) {
 					setFocus( /** @type {HTMLElement} */ ( elementToFocus ) );

```

Reply:
✅ Updated. This was very helpful.